pkg/api/types.go

```diff
@@ -451,8 +452,16 @@ func (l *VolumeList) Set(value string) error {
 	default:
 		return fmt.Errorf("invalid source:path definition")
 	}
+
 	s := VolumeSpec{Source: filepath.Clean(mount[0]), Destination: filepath.Clean(mount[1])}
-	if IsInvalidFilename(s.Source) || IsInvalidFilename(s.Destination) {
+
+	// Trim possible :z/:Z flags before sanity check
+	destination := s.Destination
+	if strings.HasSuffix(destination, ":z") || strings.HasSuffix(destination, ":Z") {
```

Comment:
After the rebase it doesn't look like we need to check for specific flags, so Docker should return an error if we use a mode it doesn't support